presto-main/src/main/java/com/facebook/presto/sql/planner/SortExpressionExtractor.java

```diff
@@ -13,26 +13,30 @@
  */
 package com.facebook.presto.sql.planner;
 
+import com.facebook.presto.sql.ExpressionUtils;
 import com.facebook.presto.sql.tree.AstVisitor;
 import com.facebook.presto.sql.tree.ComparisonExpression;
 import com.facebook.presto.sql.tree.Expression;
-import com.facebook.presto.sql.tree.LogicalBinaryExpression;
 import com.facebook.presto.sql.tree.Node;
 import com.facebook.presto.sql.tree.SymbolReference;
 import com.google.common.collect.ImmutableList;
 
+import java.util.List;
 import java.util.Optional;
 import java.util.Set;
 
-import static com.facebook.presto.sql.tree.LogicalBinaryExpression.Type.AND;
 import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
 import static com.google.common.collect.ImmutableSet.toImmutableSet;
 import static java.util.Collections.singletonList;
+import static java.util.Comparator.comparing;
 import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.groupingBy;
```

Comment:
cmt msg "Consider different sort expressions in SortExpressionExtractor" -- it is not quite clear. Maybe

"Extract sort expressions from complex join filters" ?